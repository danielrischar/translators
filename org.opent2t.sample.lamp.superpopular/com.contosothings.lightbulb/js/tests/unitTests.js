var test = require('ava');
var translatorPath = require('path').join(__dirname, '..');
var runLampTests = require('opent2t-device-lamp/lampTests');
var deviceData = require('./devicedata');
var testData = require('./testdata');
var MockHub = require('opent2t-device-mockcontosothingshub/mockContosoThingsHub');
var mockHub = new MockHub(deviceData);

var settings = {
    createTranslator: mockHub.createTranslator(translatorPath, deviceData.base_state.id),
    test: test,
    setTestData: mockHub.setTestData,
    expectedExceptions : testData.expected_exceptions
};

// Run standard lamp unit tests
runLampTests(settings);