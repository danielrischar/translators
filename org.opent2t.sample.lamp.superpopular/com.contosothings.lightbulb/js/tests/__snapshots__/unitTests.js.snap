exports[`GetPlatform 1`] = `"{\"opent2t\":{\"schema\":\"org.opent2t.sample.lamp.superpopular\",\"translator\":\"opent2t-translator-com-contosothings-lightbulb\",\"controlId\":\"fb8e4fef-092c-486f-80d7-502a2715dfb7\"},\"pi\":\"fb8e4fef-092c-486f-80d7-502a2715dfb7\",\"mnmn\":\"ContosoThings\",\"mnmo\":\"ctLight1\",\"n\":\"Backyard Lights\",\"rt\":[\"org.opent2t.sample.lamp.superpopular\"],\"entities\":[{\"n\":\"Backyard Lights\",\"rt\":[\"opent2t.d.light\"],\"di\":\"F8CFB903-58BB-4753-97E0-72BD7DBC7933\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/power\",\"rt\":[\"oic.r.switch.binary\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]},{\"href\":\"/dim\",\"rt\":[\"oic.r.light.dimming\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]}]}]}"`;

exports[`GetPlatformExpanded 1`] = `"{"opent2t":{"schema":"org.opent2t.sample.lamp.superpopular","translator":"opent2t-translator-com-contosothings-lightbulb","controlId":"fb8e4fef-092c-486f-80d7-502a2715dfb7"},"pi":"fb8e4fef-092c-486f-80d7-502a2715dfb7","mnmn":"ContosoThings","mnmo":"ctLight1","n":"Backyard Lights","rt":["org.opent2t.sample.lamp.superpopular"],"entities":[{"n":"Backyard Lights","rt":["opent2t.d.light"],"di":"F8CFB903-58BB-4753-97E0-72BD7DBC7933","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/power","rt":["oic.r.switch.binary"],"if":["oic.if.a","oic.if.baseline"],"id":"power","value":false},{"href":"/dim","rt":["oic.r.light.dimming"],"if":["oic.if.a","oic.if.baseline"],"id":"dim","dimmingSetting":0,"range":[0,100]}]}]}"`;
